overlord/devicestate/devicemgr.go

```diff
@@ -400,6 +400,17 @@ func (m *DeviceManager) ensureOperational() error {
 	return nil
 }
 
+func getPopulateStateFromSeedOptions() (*populateStateFromSeedOptions, error) {
+	modeEnv, err := boot.ReadModeenv("")
```

Comment:
we should store the full modeEnv on DeviceManager.modeEnv when we read it first, the this can be a method on DeviceManager